src/Avalonia.Controls/ComboBox.cs

```diff
@@ -262,6 +262,21 @@ protected override void OnApplyTemplate(TemplateAppliedEventArgs e)
             _popup.Closed += PopupClosed;
         }
 
+        protected override void OnGotFocus(GotFocusEventArgs e)
+        {
+            if (e.NavigationMethod == NavigationMethod.Tab
+                || e.NavigationMethod == NavigationMethod.Directional)
+            {
+                PseudoClasses.Set(":focusvisual", true);
```

Comment:
@grokys  @danwalmsley
ComboBox has custom focus style changing its template, so we can't override FocusAdorner.
As I mentioned above, :focus will be set with any kind of focus, so we can't use it for ComboBox.
Because of that I have added new :focusvisual pseudoclass with same condition as it is for FocusAdorner.
Not sure tho if it is good workaround.
Probably we need some universal way, so we can use it with other controls.

Reply:
Firefox has a [`:-moz-focusring`](https://developer.mozilla.org/en-US/docs/Web/CSS/:-moz-focusring) pseudoclass (that is apparently in the process of being standardized to `:focus-ring`) that seems to do what we want. We should consider adding this pseudoclass or similar to Avalonia I think.

Reply:
Good idea.
But "focus-visible" class is blocked by https://github.com/AvaloniaUI/Avalonia/issues/3552